src/Common/tests/Collections/IListTest.cs

```diff
@@ -0,0 +1,1692 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using Xunit;
+// ReSharper disable InconsistentNaming
+
+namespace Tests.Collections
+{
+    public enum CollectionOrder
```

Comment:
I don't want to put this is a separate file because these will be consumed by multiple projects. I want to make adding these to another project easy. I will move it to `IEnumerableTest`.